src/resolver.rs

```diff
@@ -0,0 +1,490 @@
+                if let Some(lhs) = ctx.call {
+                    //special context for left hand side
+                    self.visit_statement(&ctx.with_pou(lhs), left);
+                } else {
+                    self.visit_statement(ctx, left);
+                }
+
+            }
+            Statement::CallStatement {
+                parameters,
+                operator,
+                ..
+            } => {
+                self.visit_statement(ctx, operator);
+                let operator_type_name = self
+                    .annotation_map
+                    .get_type(operator, self.index)
+                    .get_name();
+                if let Some(s) = parameters.as_ref() {
+                    let ctx = ctx.with_call(operator_type_name);
+                    self.visit_statement(&ctx, s);
+                }
+
+                if let Some(return_type) = self
+                    .index
+                    .find_return_type(operator_type_name)
+                    .and_then(|it| self.index.find_effective_type(it))
+                {
+                    self.annotation_map
+                        .annotate_type(statement, return_type.get_name());
+                }
+            }
+            _ => {
+                self.visit_statement_literals(ctx, statement);
+            }
+        }
+    }
+
+    /// annotate a literal statement
+    fn visit_statement_literals(&mut self, ctx: &VisitorContext, statement: &Statement) {
+        match statement {
+            Statement::LiteralBool { .. } => {
+                self.annotation_map.annotate_type(statement, BOOL_TYPE)
+            }
+            Statement::LiteralString { .. } => {
+                self.annotation_map.annotate_type(statement, STRING_TYPE);
+            }
+            Statement::LiteralInteger { value, .. } => {
+                self.annotation_map
+                    .annotate_type(statement, get_int_type_name_for(*value));
+            }
+            Statement::LiteralTime { .. } => {
+                self.annotation_map.annotate_type(statement, TIME_TYPE)
+            }
+            Statement::LiteralTimeOfDay { .. } => {
+                self.annotation_map
+                    .annotate_type(statement, TIME_OF_DAY_TYPE);
+            }
+            Statement::LiteralDate { .. } => {
+                self.annotation_map.annotate_type(statement, DATE_TYPE);
+            }
+            Statement::LiteralDateAndTime { .. } => {
+                self.annotation_map
+                    .annotate_type(statement, DATE_AND_TIME_TYPE);
+            }
+            Statement::LiteralReal { .. } => {
+                //TODO when do we need a LREAL literal?
+                self.annotation_map.annotate_type(statement, REAL_TYPE);
+            }
+            Statement::LiteralArray {
+                elements: Some(elements),
+                ..
+            } => {
+                self.visit_statement(ctx, elements.as_ref());
```

Comment:
This is still a problem. how can i guess a name for the resulting array-type?

Reply:
How about this?
Array size is easy to find by the number of elements
The array is always called __ARRAY__X__Y__OF__TYPE
X and Y are dim sizes.

Reply:
ok so I would need to change the way we name generated Types ... it will postpone this commit. maybe we add it as an open issue